lib/fund_america/api.rb

```diff
@@ -5,19 +5,15 @@ class << self
       # This method is called from each end point method to make API requests
       # using HTTParty gem. Takes the method, uri and options as input
       # Handles response and errors
-      def request method, uri, options={}
-        options = FundAmerica.basic_auth.merge!({:body => options})
+      def request(method, uri, options = {})
+        options = FundAmerica.basic_auth.merge!(body: options)
         uri = FundAmerica.base_uri + uri unless uri.include?('test_mode')
-        response = HTTParty.send(method, uri, options)
-        parsed_response = JSON.parse(response.body)
-        if response.code.to_i == 200
-          # Returns parsed_response - a hash of response body
-          # if response is successful
-          parsed_response
-        else
-          # Raises error if the response is not sucessful
-          raise FundAmerica::Error.new(parsed_response, response.code.to_i)
-        end
+        @response = HTTParty.send(method, uri, options)
+
+        raise_wrong_status_error unless success_response
+
+        parsed_response
+      rescue JSON::ParserError
```

Comment:
@Ununuk have you tested that with `investments/delete`? We need to make sure that it won't break anything once we receive `null` in the request's body.

Reply:
I added rescue for it, but now it seems to be working. When I want to delete investment x times then I have investment in the response (no json error).

Reply:
There are 2 cases for it:
1. When I run the console and then call delete then I got nil.
2. When I run the console, call list on investments and then call delete then I got investment.
```
~/RubyMineProjects/BankToTheFuture(AT-1755/prepare_fund_america_gem_for_bf ✔) rails c
Running via Spring preloader in process 48391
Loading development environment (Rails 4.2.11.1)
[1] pry(main)> FundAmerica.api_key = 'LbM2-I0dggMdEP5EtMK1Ns2ksqxOkCfV'
=> "LbM2-I0dggMdEP5EtMK1Ns2ksqxOkCfV"
[2] pry(main)> FundAmerica::Investment.delete('db_2F_4pTf2W3dtDx235Rw')
=> nil
[3] pry(main)> FundAmerica::Investment.delete('db_2F_4pTf2W3dtDx235Rw')
=> nil
[4] pry(main)> FundAmerica::Investment.list
=> {"object"=>"resource_list",
 "total_resources"=>1,
 "page"=>1,
 "per"=>25,
 "more"=>false,
 "resources"=>
  [{"object"=>"investment",
    "id"=>"db_2F_4pTf2W3dtDx235Rw",
    "url"=>"https://sandbox.fundamerica.com/api/investments/db_2F_4pTf2W3dtDx235Rw",
    "offering_url"=>"https://sandbox.fundamerica.com/api/offerings/AflShkTFRm6neTWJe6UXcA",
    "administration_fee"=>"0.0",
    "aml_check_investor"=>true,
    "aml_exception"=>false,
    "amount"=>"1000.0",
    "amount_in_escrow"=>"0.0",
    "amount_received"=>"0.0",
    "amount_refunded"=>"0.0",
    "background_check_url"=>"https://sandbox.fundamerica.com/api/background_checks/6YvnddBsRiyRtKHMNltlNA",
    "bank_reference"=>"FA4Q6M",
    "bank_wire_info"=>
     "Bank: Western Alliance Bank\nOne East Washington Street\nPhoenix AZ 85004\n1 (602) 629-1776\nABA/Routing #: 122105980\n\nBeneficiary: \n\nAccount #: \nREF: FA4Q6M",
    "brokerage_fee"=>"0.0",
    "ceremonial_certificate_url"=>nil,
    "check_mailing_address"=>"Prime Trust\n10890 S Eastern Ave\nSuite 114\nHenderson, NV 89502",
    "check_mailing_instructions"=>
     "Please make checks payable to \"Prime Trust as Escrow Agent for Offering 1565081501\" and include a note with your name, phone number, email address, and investor ID (FA4Q6M) in case we have any questions.",
    "cleared"=>false,
    "clearing_failures"=>["aml_pending", "funds_not_cleared"],
    "commission_fee"=>"0.0",
    "concession_fee"=>"0.0",
    "confirm_accreditation"=>false,
    "created_at"=>"2019-08-20T09:06:51.955Z",
    "data"=>{},
    "debt_face_value"=>nil,
    "debt_interest_rate"=>nil,
    "debt_maturity_date"=>nil,
    "debt_par_value"=>"1000.0",
    "disbursed_at"=>nil,
    "entity_url"=>"https://sandbox.fundamerica.com/api/entities/RbYYHi4VQke1X-LmxU6Ejw",
    "equity_share_count"=>"1000.0",
    "equity_share_price"=>"1.0",
    "face_value"=>nil,
    "financial_adviser_name"=>nil,
    "financial_advisor_name"=>nil,
    "funds_disbursable"=>0,
    "funds_transfer_method"=>"wire",
    "in_escrow_at"=>nil,
    "incomplete"=>false,
    "incomplete_reason"=>nil,
    "investor_url"=>"https://sandbox.fundamerica.com/api/investors/7TlEcrZDQV2ibPELPfFrSQ",
    "kyc_by_client"=>true,
    "par_value"=>"1000.0",
    "payment_method"=>"wire",
    "payment_reference"=>"FA4Q6M",
    "promo_code"=>nil,
    "referrer"=>nil,
    "refunded_at"=>nil,
    "remittance_details"=>
     {"bank_address"=>"One East Washington Street\nPhoenix AZ 85004",
      "bank_name"=>"Western Alliance Bank",
      "bank_phone"=>"1 (602) 629-1776",
      "routing_number"=>"122105980",
      "swift_code"=>"BBFXUS6S",
      "account_number"=>"00015738912",
      "beneficiary_name"=>"Prime Trust as Escrow Agent for Offering 1565081501",
      "beneficiary_address"=>"10890 S Eastern Ave\nSuite 114\nHenderson, NV 89502",
      "reference"=>"FA4Q6M"},
[5] pry(main)> FundAmerica::Investment.delete('db_2F_4pTf2W3dtDx235Rw')
=> {"object"=>"resource_list",
 "total_resources"=>1,
 "page"=>1,
 "per"=>25,
 "more"=>false,
 "resources"=>
  [{"object"=>"investment",
    "id"=>"db_2F_4pTf2W3dtDx235Rw",
    "url"=>"https://sandbox.fundamerica.com/api/investments/db_2F_4pTf2W3dtDx235Rw",
    "offering_url"=>"https://sandbox.fundamerica.com/api/offerings/AflShkTFRm6neTWJe6UXcA",
    "administration_fee"=>"0.0",
    "aml_check_investor"=>true,
    "aml_exception"=>false,
    "amount"=>"1000.0",
    "amount_in_escrow"=>"0.0",
    "amount_received"=>"0.0",
    "amount_refunded"=>"0.0",
    "background_check_url"=>"https://sandbox.fundamerica.com/api/background_checks/6YvnddBsRiyRtKHMNltlNA",
    "bank_reference"=>"FA4Q6M",
    "bank_wire_info"=>
     "Bank: Western Alliance Bank\nOne East Washington Street\nPhoenix AZ 85004\n1 (602) 629-1776\nABA/Routing #: 122105980\n\nBeneficiary: \n\nAccount #: \nREF: FA4Q6M",
    "brokerage_fee"=>"0.0",
    "ceremonial_certificate_url"=>nil,
    "check_mailing_address"=>"Prime Trust\n10890 S Eastern Ave\nSuite 114\nHenderson, NV 89502",
    "check_mailing_instructions"=>
     "Please make checks payable to \"Prime Trust as Escrow Agent for Offering 1565081501\" and include a note with your name, phone number, email address, and investor ID (FA4Q6M) in case we have any questions.",
    "cleared"=>false,
    "clearing_failures"=>["aml_pending", "funds_not_cleared"],
    "commission_fee"=>"0.0",
    "concession_fee"=>"0.0",
    "confirm_accreditation"=>false,
    "created_at"=>"2019-08-20T09:06:51.955Z",
    "data"=>{},
    "debt_face_value"=>nil,
    "debt_interest_rate"=>nil,
    "debt_maturity_date"=>nil,
    "debt_par_value"=>"1000.0",
    "disbursed_at"=>nil,
    "entity_url"=>"https://sandbox.fundamerica.com/api/entities/RbYYHi4VQke1X-LmxU6Ejw",
    "equity_share_count"=>"1000.0",
    "equity_share_price"=>"1.0",
    "face_value"=>nil,
    "financial_adviser_name"=>nil,
    "financial_advisor_name"=>nil,
    "funds_disbursable"=>0,
    "funds_transfer_method"=>"wire",
    "in_escrow_at"=>nil,
    "incomplete"=>false,
    "incomplete_reason"=>nil,
    "investor_url"=>"https://sandbox.fundamerica.com/api/investors/7TlEcrZDQV2ibPELPfFrSQ",
    "kyc_by_client"=>true,
    "par_value"=>"1000.0",
    "payment_method"=>"wire",
    "payment_reference"=>"FA4Q6M",
    "promo_code"=>nil,
    "referrer"=>nil,
    "refunded_at"=>nil,
    "remittance_details"=>
     {"bank_address"=>"One East Washington Street\nPhoenix AZ 85004",
      "bank_name"=>"Western Alliance Bank",
      "bank_phone"=>"1 (602) 629-1776",
      "routing_number"=>"122105980",
      "swift_code"=>"BBFXUS6S",
      "account_number"=>"00015738912",
      "beneficiary_name"=>"Prime Trust as Escrow Agent for Offering 1565081501",
      "beneficiary_address"=>"10890 S Eastern Ave\nSuite 114\nHenderson, NV 89502",
      "reference"=>"FA4Q6M"},
```